app/src/main/java/org/simple/clinic/di/network/RetrofitModule.kt

```diff
@@ -54,4 +55,22 @@ class RetrofitModule {
         .baseUrl(endpoint)
         .build()
   }
+
+  @Provides
+  @AppScope
+  @Named("for_deployment")
+  fun retrofitForDeployment(
+      deployment: Deployment,
+      commonRetrofitBuilder: Retrofit.Builder
+  ): Retrofit {
+    // Since the endpoint is not under our control, and is defined at the server level,
+    // this is a safety check that will generate the right endpoint regardless of whether the
+    // endpoint defined in the manifest has a trailing slash or not.
+    val baseUrl = deployment.endPoint.toString().removeSuffix("/")
```

Comment:
I didn't understand why we're checking this? So there is no possibility of getting the slash in the url that we receive from server but we're checking as a safety precaution?

Reply:
Retrofit expects the base url to end with one trailing slash. Since we are getting this from server and don't know how it's defined in manifest, we are removing all ending slashes and adding just one.